### PR TITLE
Add new style

### PR DESCRIPTION
**Pull request by darleendurand**
-Added form to add new pizza style
-Added errorlist page to handle for errors on the form
-Added pathing for the form in homecontroller and navbar
-Added a @PostMapping 
-Added new db query to check if pizza style already exists before creating new one
-Added custom error handling models
-Added validation checks

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,8 +84,13 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-validation</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>jakarta.validation</groupId>
+			<artifactId>jakarta.validation-api</artifactId>
+			<version>2.0.2</version>
+		</dependency>
```
[the remark this answers is not in the document]
Comment (darleendurand) [reply pom.xml]: Removed this dependency

File: src/main/java/com/launchacademy/reviews/controllers/PizzaStylesApiV1Controller.java [hunk 2]
```diff
@@ -26,4 +26,17 @@ public PizzaStylesApiV1Controller(PizzaStyleService pizzaStyleService) {
     pizzaStylesMap.put("pizzaStyles", pizzaStyleService.findAll());
     return pizzaStylesMap;
   }
+
+  @PostMapping
+  public Map<String, PizzaStyle> addPizzaStyle(@RequestBody @Valid PizzaStyle pizzaStyle) {
```
[the remark this answers is not in the document]
Comment (darleendurand) [reply src/main/java/com/launchacademy/reviews/controllers/PizzaStylesApiV1Controller.java]: Added BindingResult to check for and return errors if present

File: src/main/java/com/launchacademy/reviews/controllers/PizzaStylesApiV1Controller.java [hunk 3]
```diff
@@ -28,6 +36,23 @@ public PizzaStylesApiV1Controller(PizzaStyleService pizzaStyleService) {
...
+      return customError.handleBindingErrors(bindingResult);
+    } else {
+      Map<String, PizzaStyle> newStyle = new HashMap<>();
+      List<PizzaStyle> checkForType = pizzaStyleService.findByNameIgnoreCase(pizzaStyle.getName());
+      if (checkForType.isEmpty()) {
+        pizzaStyleService.save(pizzaStyle);
+        newStyle.put("pizzaStyle", pizzaStyle);
+        return newStyle;
+      } else {
+        return customError.alreadyExists();
```
Comment (safang99) [inline src/main/java/com/launchacademy/reviews/controllers/PizzaStylesApiV1Controller.java:51]: very nice 🙌

**Review by safang99 — APPROVED** [review]
Fantastic work and nice check for uniqueness!

**Pull request merged**